brainscore_vision/metrics/maniquet2024_metrics/metric.py

```diff
@@ -0,0 +1,440 @@
+        correlation_scores = []
+
+        # Create confusion matrices for all participants
+        confmats = {sub:
+            confusion_matrix(
+            y_true=assembly.loc[assembly['subject_id'] == sub]["image_label"],
+            y_pred=assembly.loc[assembly['subject_id'] == sub]["prediction"],
+            labels=labels,
+        ) for sub in subjects}
+
+        # Run 100 split-half permutations
+        for i in range(100):
+
+            # Get random half of participants
+            half_subjects = random.sample(list(subjects), k=len(subjects) // 2)
+
+            # Get average of confusion matrices for the two groups
+            part_one = np.mean([confmats[subject] for subject in half_subjects], axis=0)
+            part_two = np.mean([confmats[subject] for subject in subjects if subject not in half_subjects], axis=0)
+
+            # Compute Pearson correlation between the two confusion matrices
+            correlation_score = pearsonr(
+                self._rollout_matrix(part_one),
+                self._rollout_matrix(part_two),
+            )[0]
+            correlation_scores.append(correlation_score)
+
+        # Average split-half correlations as a measure of reliability
+        ceiling = np.mean(correlation_scores)
+
+        return ceiling
+
+    def __call__(
+        self, probabilities: BehavioralAssembly, human_assembly: BehavioralAssembly
+    ) -> Score:
+        """
+        Computes the correlation between model and human confusion matrices normalized by the ceiling.
+
+        Args:
+            probabilities (BehavioralAssembly): Model's predicted probabilities.
+            human_assembly (BehavioralAssembly): Human baseline responses.
+
+        Returns:
+            Score: The normalized correlation score as a performance metric.
+        """
+        assert sorted(set(probabilities.choice.values)) == sorted(
+            set(human_assembly.image_label.values)
+        )
+
+        # Extract labels from the model probabilities.
+        y_true, y_pred = self._label_from_probability(probabilities)
+
+        # Calculate the model's confusion matrix.
+        dnn_confmat = confusion_matrix(
+            y_true=y_true, y_pred=y_pred, labels=probabilities.choice.values
+        )
+
+        # Calculate the human confusion matrix.
+        human_confmat = confusion_matrix(
+            y_true=human_assembly["image_label"],
+            y_pred=human_assembly["prediction"],
+            labels=probabilities.choice.values,
+        )
+
+        # Compute the Pearson correlation between the model and human confusion matrices.
+        correlation_score = pearsonr(
+            self._rollout_matrix(human_confmat), self._rollout_matrix(dnn_confmat)
+        )[0]
+        ceiling = self._ceiling(human_assembly, precomputed=True)
+
+        # Normalize by ceiling
+        score = Score(correlation_score)
+        score.attrs["normalized"] = correlation_score / ceiling
+        score.attrs["ceiling"] = ceiling
```

Comment:
conforming with the structure of other benchmarks, let's keep this in the benchmark itself

```suggestion
        score = Score(correlation_score)
```